.travis.yml

```diff
@@ -34,6 +43,9 @@ install:
   - /usr/libexec/java_home -v 1.7 -F -V
   - java -Xmx32m -version && javac -J-Xmx32m -version
   - pushd j2objc-gradle
+  - echo For branch $TRAVIS_BRANCH
+  # Update the plugin submodule to latest HEAD if in the appropriate branch
+  - if [[ $TRAVIS_BRANCH == pluginHead ]]; then git pull origin master; fi
```

Comment:
I think master should refer to both the master in e2e and master in the plugin. Then get rid of "pluginHead" and replace it with "pluginRelease", so that it's use is more clear. You can then make it update to head and then grab the latest tag or HEAD (master) as appropriate. Replace the current line with:

```
git pull origin master
if [[ $TRAVIS_BRANCH == pluginLatestVersion ]]; then
    TAG=`git tag -l | sort | tail -n 1`
    echo Plugin Tag: $TAG
    git checkout tags/TAG
fi
```

The sort is so that if the latest tag was a hotfix on old release (e.g. 0.4.2.1), it will still use the latest version for builds (v0.5.0), even though it was an older tag. We could then get really fancy and add another section again:

```
if [[ $TRAVIS_BRANCH == pluginLatestTag ]]; then
    LATEST_VERSION=`git tag -l | sort | tail -n 1`
    LATEST_TAG=`git tag -l | tail -n 1`
    if [[ $LATEST_TAG != $LATEST_VERSION ]]; then
        echo Plugin Tag: $LATEST_TAG
        git checkout tags/LATEST_TAG
    else
        # LATEST_TAG == LATEST_VERSION, so testing isn't needed
        # SOMEHOW STOP THE BUILD AND PASS EVERYTHING
    fi
fi
```